pkg/manager/manager.go

```diff
@@ -290,104 +299,255 @@ func (m MediaManager) AddMovieToLibrary(ctx context.Context, request AddMovieReq
-	for i, indexer := range indexers {
-		indexerIds[i] = indexer.ID
+
+	snapshot := newReconcileSnapshot(indexers, dcs)
+
+	wg.Add(1)
+	go m.ReconcileMissingMovies(ctx, &wg, snapshot)
+
+	wg.Add(1)
+	go m.ReconcileUnreleasedMovies(ctx, &wg, snapshot)
+
+	wg.Wait()
+	return nil
+}
+
+func (m MediaManager) ReconcileMissingMovies(ctx context.Context, wg *sync.WaitGroup, snapshot *ReconcileSnapshot) error {
+	defer wg.Done()
+
+	if snapshot == nil {
+		return fmt.Errorf("snapshot is nil")
 	}
 
 	movies, err := m.storage.ListMoviesByState(ctx, storage.MovieStateMissing)
 	if err != nil {
 		return fmt.Errorf("couldn't list movies during reconcile: %w", err)
 	}
 
-	for _, mov := range movies {
-		if mov.MovieMetadataID == nil || mov.QualityProfileID == 0 || mov.Monitored == 0 {
-			continue
-		}
-
-		if mov.MovieFileID != nil {
-			// TODO: this probably needs to check if the existing file meets the quality cutoff
-			continue
-		}
+	log := logger.FromCtx(ctx)
 
-		det, err := m.storage.GetMovieMetadata(ctx, table.MovieMetadata.ID.EQ(sqlite.Int32(*mov.MovieMetadataID)))
+	for _, movie := range movies {
+		err = m.reconcileMissingMovie(ctx, movie, snapshot)
 		if err != nil {
-			log.Debugw("failed to find movie metadata", "meta_id", *mov.MovieMetadataID)
-			continue
+			log.Warn("failed to reconcile movie", zap.Error(err))
 		}
+	}
 
-		profile, err := m.storage.GetQualityProfile(ctx, int64(mov.QualityProfileID))
-		if err != nil {
-			log.Warnw("failed to find movie qualityprofile", "quality_id", mov.QualityProfileID)
-			continue
-		}
+	return nil
+}
 
-		releases, err := m.SearchIndexers(ctx, indexerIds, MOVIE_CATEGORIES, det.Title)
-		if err != nil {
-			log.Debugw("failed to search indexer", "indexers", indexerIds, zap.Error(err))
-			continue
-		}
+func (m MediaManager) reconcileMissingMovie(ctx context.Context, movie *storage.Movie, snapshot *ReconcileSnapshot) error {
+	log := logger.FromCtx(ctx)
+	log = log.With("movie id", movie.ID)
 
-		log.Debugw("releases for consideration", "releases", len(releases))
-		releases = slices.DeleteFunc(releases, rejectReleaseFunc(ctx, det, profile, protocolsAvailable))
-		log.Debugw("releases after rejection", "releases", len(releases))
-		if len(releases) == 0 {
-			// move on the next movie
-			continue
-		}
+	if movie.MovieMetadataID == nil {
+		log.Warn("movie metadata id is nil, skipping reconcile")
+		return nil
+	}
 
-		slices.SortFunc(releases, sortReleaseFunc())
-		chosenRelease := releases[len(releases)-1]
+	if movie.QualityProfileID == 0 {
+		log.Warn("movie quality profile id is nil, skipping reconcile")
+		return nil
+	}
 
-		log.Infow("found release", "title", chosenRelease.Title, "proto", *chosenRelease.Protocol)
+	if movie.Monitored == 0 {
+		log.Warn("movie is not monitored, skipping reconcile")
+	}
 
-		downloadRequest := download.AddRequest{
-			Release: chosenRelease,
-		}
+	if movie.MovieFileID != nil {
+		log.Warn("movie file id is nil, skipping reconcile")
```

Comment:
This message doesn't match the condition right? Is it a warning if a movie already has a file in this loop? Would the status ever get updated correctly if not here?

Reply:
I added a TODO here, but I feel like we should never want to end up with a state of missing BUT having a movie file id already on disk.. something to look at for the future. Feel like we need somewhere to track these TODOs outside of code comments too..

but yes your comment was correct, the log didn't match the logic, but the logic was correct. I updated the log and set it at the debug level